### ISSUE TITLE
Subscription Center & User Plans

### ISSUE THREAD

**Issue by dularion** [dularion/streama#602]
as pointed out by @IzzyMichiel PayPal would be great as payment
I think this requires some concept work. 

- What kinds of user plans are there supposed to be? 
- should certain plans limit the amount of active devices used per user? 
- should certain features only be available for certain user plans, such as direct download? 

What are your thoughts guys?

**Comment by dularion** [comment]
(i rephrased the title and first comment a bit)

**Comment by junnotpiper** [comment]
Oh I get it, I love the idea of adverts. The download feature will be lovely if users can't access the files only streama would be able to read

**Comment by dularion** [comment]
https://github.com/dularion/streama/issues/231

**Comment by dularion** [comment]
I think that sounds like a cool idea, and some docs to go along with it. Also, maybe we can add a backend where users can dynamically create "plans" by checking and unchecking a couple of fixed features.

**Comment by Jeronimo95** [comment]
How about user editable groups? That way from the API you could place people into pre-configured groups. And from the stream admin page you could use the groups too, so it's not API specific.

Streama could ship with some default groups like we have now: Admin/Content editor/User.

**Comment by benno1214** [comment]
As far to add subscriptions etc, a idea to add like custom java scripts code to the site in settnings menu, like changing colors, changing layout etc, don't know where to change it my self, would be cool to do that directly in the website settnings.

keep up the good work, if you take crypto im gladly donating zcash.

-benno

**Comment by Jeronimo95** [comment]
@tipcircle Like I said previously: I don't think streama should be the one that actually integrates the payment processors. We would provide an API or Extention interface that would allow admins integrate whatever payment processor they want. That way you wouldn't be limited any particular payment processor.

@benno1214 Good idea. We could have some more appearance customisation options in the settings menu.

**Comment by pmjm** [comment]
I agree that "user groups" would be handy, so you can configure a few types of accounts. Expiration dates with optional displayable warnings as the expiration approaches would be helpful.

As far as a subscription system itself, that might be beyond the scope of the project. I'm not opposed to these types of features but not at the expense of other things that are needed more urgently.

**Comment by playdream** [comment]
Personally i think you should add users groups where we can create the different types of users e.g 
Free users 
Registered users
Premium users

we can set if these groups can register ect 
(also which file users stored in while the app is running what to create a external register while this in dev) 

thank you

**Comment by Exe18** [comment]
Sinceramente, creo que es muy buena idea, que se pueda agregar un sistema de suscripciones es una opción muy interesante en el caso de que otros usuarios estén interesados en su biblioteca multimedia para mi seria muy útil, desde ya muchas gracias a los desarrolladores de la  misma muy buen trabajo, espero que esto se vuelva una mejora en el sistema !!!

**Comment by HannahRy** [comment]
We are not going to be integrating payment systems into Streama. There are many reasons for this:

1. It's very hard to maintain secure payments systems, even if intergaing with an API. 
2. If we were to integrate with an API, we wouldn't be able to have just one, as not all payment APIs would be suitable for all users. This would add further complexity to the project.
3. The legality of people charging for content. This is the same reason we won't be implementing any sort of downloading.


If you are using this commercially to serve content you own the rights to - then you will need to add these features yourself.

**Comment by adamgiesinger** [comment]
> This is the same reason we won't be implementing any sort of downloading.
Isn't there an option that allowes users to download?

**Comment by HannahRy** [comment]
I mean implementing downloading from torrents/usenet/etc.

**Comment by Exe18** [comment]
La opción de descarga ya está implementada en las versión recientes y en el caso de la suscripción  creo que muchos están de acuerdo con esta idea y si en caso del sistema de pago sería una función aparte que el usuario pueda cambiar y agregar su propio sistema de pago 

Gracias a aquellos que si quieren innovar

**Comment by adamgiesinger** [comment]
> I mean implementing downloading from torrents/usenet/etc.

Oh sorry, my bad.

**Comment by glinskicwb1988** [comment]
Any updates on user signup ? 

I don't think that ALL the features (like payment gateway, etc..) must be on order to launch this feature, maybe its because of this that is taking so long.

A link to the user signup by him/her self would be great already.
